### DISCUSSION TITLE
Change Input Suggester dynamically

### DISCUSSION THREAD

**Discussion by vpet98** [Textualize/textual#3915] [Q&A] [answered]
Is it possible to change the suggestion list of an Input's suggester as the user types? For example, like below:
```python
def on_input_changed(self, event: Input.Changed) -> None:
    if event.value and event.value[-1] == '/':
        event.input.suggester=SuggestFromList(os.listdir(event.value), case_sensitive=False)
```
The goal is every time the user writes a '/' they get suggestions from the available files in the path they have specified so far.

When I tried it seemed that the suggester cannot be changed this way.

**Comment by davep** [comment]
Rather than attempt to change the suggester on the fly like that, it would make more sense for you to implement a filesystem content suggester, implementing your own `get_suggestion` method that completes for the current value.

If it helps, while it isn't for a similar result, [here's some code where I wrote my own suggester that suggests tags for a tag input field, taking into account tags that have already been used, etc](https://github.com/davep/tinboard/blob/main/tinboard/suggestions/tags.py).

**Reply by vpet98** [reply]
But doesn't that mean that you provide all the tags when initializing the Suggester? While this cannot be done for a whole file system.

**Reply by davep** [reply, marked as answer]
I wouldn't get too caught up in how that happens; I was showing more the pattern for how you'd subclass the base class and implement your own `get_suggestion`. A suggester for a filesystem should be pretty straightforward; remember that `get_suggestion` each time there's a change to the `Input`, so you should only need take the value it passes, parse out the directory part (nice and easy if you use `pathlib.Path` for example), and then return the best suggestion from within that directory.
